packages/combobox/src/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,740 @@
+    contentRef?: React.RefObject<HTMLDivElement>;
+  }
+>({
+  onValueChange: () => {},
+  contentId: "",
+  triggerId: "",
+  open: false,
+  setOpen: () => {},
+  onOpenToggle: () => {},
+});
+
+const Root = ({
+  modal = true,
+  closeOnSelect = true,
+  open: openProp,
+  onOpenChange: onOpenChangeProp,
+  defaultOpen: defaultOpenProp,
+  value,
+  onValueChange,
+  errored,
+  placeholder,
+  layout,
+  ...props
+}: RootProps) => {
+  const contentId = React.useId();
+  const triggerId = React.useId();
+  const triggerRef = React.useRef(null);
+  const searchRef = React.useRef(null);
+  const contentRef = React.useRef<HTMLDivElement>(null);
+
+  const [searchQuery, setSearchQuery] = React.useState<string>("");
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpenProp,
+    onChange: onOpenChangeProp,
+  });
+
+  const onOpenToggle = React.useCallback(() => {
+    setOpen((prevOpen) => !prevOpen);
+  }, [setOpen]);
+
+  React.useEffect(() => {
+    // If the combobox is closed clear
+    // the search query
+    if (!open) {
+      setSearchQuery("");
+    }
+  }, [open]);
+
+  return (
+    <ComboboxContext.Provider
+      value={{
+        contentId,
+        triggerId,
+        value,
+        onValueChange,
+        placeholder,
+        open,
+        setOpen,
+        onOpenToggle,
+        closeOnSelect,
+        searchQuery,
+        setSearchQuery,
+        triggerRef,
+        searchRef,
+        contentRef,
+        errored,
+        layout,
+      }}
+    >
+      <TelegraphMenu.Root
+        open={open}
+        onOpenChange={setOpen}
+        modal={modal}
+        {...props}
+      />
+    </ComboboxContext.Provider>
+  );
+};
+
+type TriggerTagProps = {
+  value: string;
+  label?: string;
+};
+
+const TriggerTag = ({ label, value, ...props }: TriggerTagProps) => {
+  const context = React.useContext(ComboboxContext);
+
+  return (
+    <Tag.Root
+      size="1"
+      as={motion.span}
+      initial={{ opacity: 0, transform: "scale(0.8)" }}
+      animate={{ opacity: 1, transform: "scale(1)" }}
+      exit={{ opacity: 0, transform: "scale(0.5)" }}
+      layout="position"
+      transition={{
+        duration: 0.2,
+        type: "spring",
+        bounce: 0,
+        layout: {
+          duration: 0.05,
+          type: "spring",
+          bounce: 0,
+        },
+      }}
+      {...props}
+    >
+      <Tag.Text>{label || value}</Tag.Text>
+      <Tag.Button
+        as="span"
+        role="button"
+        icon={{ icon: Lucide.X, alt: `Remove ${value}` }}
+        onClick={(event: React.MouseEvent) => {
+          const onValueChange = context.onValueChange as (
+            v: Array<Option>,
+          ) => void;
+          const contextValue = context.value as Array<Option>;
+          const newValue = contextValue.filter((v) => v.value !== value);
+          onValueChange(newValue);
+          event.stopPropagation();
```

Comment:
stopPropagation so that we don't open or close the combobox when removing a tag.